spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/LoggingWebMvcConfigurerSettings.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+/**
+ * Holds settings for @{@link LoggingWebMvcConfigurer}
+ * @author Chengyuan Zhao
+ */
+public class LoggingWebMvcConfigurerSettings {
+
+	public TraceIdExtractorCombination getExtractorCombination() {
+		return this.extractorCombination;
+	}
+
+	public void setExtractorCombination(
+			TraceIdExtractorCombination extractorCombination) {
+		this.extractorCombination = extractorCombination;
+	}
+
+	private TraceIdExtractorCombination extractorCombination;
```

Comment:
Why do we need this class at all to just hold a single instance variable?

Reply:
It's a convenient place to put other settings in the future. Also, it allows the starter project to have a `ConfigurationSettings` annotated type inherit from this. We did this exact same thing in the storage project to avoid having the project depend on its starter.

Reply:
But this is not `Settings`, this is just an `enum`. And again: the configuration properties is a feature of the Boot. Here in the raw level we should allow just configure beans and inject them. That is exactly the purpose of the Dependency Injection. The "Configuration by Convention" is a part of other effort - Spring Boot.